ai/STA/Strategy/strategy.py

```diff
@@ -21,23 +21,24 @@ def __init__(self, p_game_state: GameState):
         """
         assert isinstance(p_game_state, GameState)
         self.game_state = p_game_state
-        self.roles = Role.as_list()
-        self.roles_graph = {r: Graph() for r in self.roles}
-        players = [p for p in self.game_state.our_team.players.values()]
 
-        role_mapping = dict(zip(self.roles, players))
-        role_mapping = self.hack_goalkeeper(role_mapping)
+        self.roles_graph = {role: Graph() for role in self.assigned_roles.keys()}
 
-        self.game_state.map_players_to_roles_by_player(role_mapping)
+    @property
+    def assigned_roles(self):
+        return self.game_state.assigned_roles
 
-    def hack_goalkeeper(self, role_mapping):
-        current_goaler = self.game_state.get_player_by_role(Role.GOALKEEPER)
-        if current_goaler is not None:
-            new_goaler = role_mapping[Role.GOALKEEPER]
-            new_goaler_old_role = self.game_state.get_role_by_player_id(new_goaler.id)
-            role_mapping[Role.GOALKEEPER] = current_goaler
-            role_mapping[new_goaler_old_role] = new_goaler
-        return role_mapping
+    @classmethod
```

Comment:
use `@abstractmethod` and `pass`

Reply:
It need to be a classmethod since it need to be call without having an instance.

Reply:
you can add both decorator ! since python 3.5 i think.

Reply:
Ok, so it's not what we want. Abstract method does not throw if super() is called. So I will keep the notImplementedError

Reply:
Its supposed to abort a class instance to be create if the method is not implemented